packages/hash/api/src/graph/system-entities.ts

```diff
@@ -17,15 +20,16 @@ export const ensureSystemEntitiesExists = async (params: {
 
   // Create system entities if they don't already exist
 
-  await HashInstanceModel.getHashInstanceModel(graphApi).catch(
-    async (error: Error) => {
-      // Create the system instance entity, if it doesn't already exist.
-      if (error instanceof NotFoundError) {
-        return await HashInstanceModel.createHashInstance(graphApi, {
+  await getHashInstance({ graphApi }, {}).catch(async (error: Error) => {
```

Comment:
> Do we want to/should we default the params to {} so we can omit it when it's not set for the function?

I couldn't work out a good way of doing this using the definition of `ImpureGraphFunction`, which is currently:
```typescript
export type ImpureGraphFunction<Parameters, ReturnType> = (
  ctx: ImpureGraphContext,
  params: Parameters,
) => ReturnType;
```

If you define the following:
```typescript
const getHashInstance: ImpureGraphFunction<undefined, Promise<HashInstance>> = ...
```
The calling `getHashInstance` would require passing `undefined` explicitly:
```
getHashInstance(ctx) // TS error: An argument for 'params' was not provided.
getHashInstance(ctx, undefined)
```
That's why I went with `{}`, because at least it's more explicit as you say. If you can think of an alternative approach which would let us do `getHashInstance(ctx)` let me know, as I'd be in favour of it.

Reply:
I think this is sensible, then. It's okay that we are explicit about no params